lib/ansible/galaxy/__init__.py

```diff
@@ -22,16 +22,56 @@
 
 from __future__ import annotations
 
+import shutil
 import os
 
 import ansible.constants as C
 from ansible import context
+from ansible.utils.display import Display
 from ansible.module_utils.common.text.converters import to_bytes
 from ansible.module_utils.common.yaml import yaml_load
 
 #      default_readme_template
 #      default_meta_template
 
+display = Display()
+
+
+def _ask_uninstall(label: str, paths: list[str], collection: bool = True, skip_prompt: bool = False) -> bool:
+    """Confirm whether the list of paths should be removed and return a boolean."""
+    if skip_prompt:
+        return True
+
+    prompt = f"Uninstalling {'collection' if collection else 'role'} {label}:"
+    prompt += (
+        "\n  Would remove:\n    " +
+        "\n    ".join(paths)
```

Comment:
Add a note saying this operation is destructive and we can not recover deleted content.

Reply:
No, pip can't recover removed files. But I've seen some other programs in the wild that have the warning spelled out.

I checked https://clig.dev, and it doesn't seem to make an explicit recommendation. Perhaps, “Saying just enough” is applicable here, though.